### PR TITLE
[UserPage] Collapse Brew Groups

### LINKED ISSUE #1797: [UserPage] Change order of Published/Private Brews
As mentioned in [this Reddit comment](https://www.reddit.com/r/homebrewery/comments/qlh2hn/bug_sorting_by_views_not_really_sorting_by/hj4p8jr), some users would like to be able to re-order the groups on the User Page, as they do not edit Published brews, which are listed at the top of the page. Having the ability to push this group to the bottom of the page would improve their user experience.

### PR DESCRIPTION
**Pull request by G-Ambatte**
This PR resolves #1797.

- mousing over headers changes mouse to a pointer
- clicking headers collapses/expands sections
- collapse state is saved to local storage on page exit
- collapse state is loaded from local storage on page load
- default state (when local storage keys don't exist) is visible

### CODE REVIEW

File: client/homebrew/pages/basePages/listPage/listPage.jsx [hunk 1]
```diff
@@ -6,29 +6,66 @@ const moment      = require('moment');
...
 			],
 			navItems : <></>
 		};
 	},
 	getInitialState : function() {
+		// HIDE ALL GROUPS UNTIL LOADED
+		const brewCollection = this.props.brewCollection.map((brewGroup)=>{
+			brewGroup.visible = false;
+			return brewGroup;
+		});
```
[the remark this answers is not in the document]
Comment (G-Ambatte) [reply client/homebrew/pages/basePages/listPage/listPage.jsx:31]: Agreed - I'll clean it up shortly.

File: client/homebrew/pages/basePages/listPage/listPage.jsx [hunk 2]
```diff
@@ -6,29 +6,66 @@ const moment      = require('moment');
...
 
+	componentDidMount : function() {
+		// SAVE TO LOCAL STORAGE WHEN LEAVING PAGE
+		window.onbeforeunload = this.saveToLocalStorage;
+
+		// LOAD FROM LOCAL STORAGE
+		if(typeof window !== 'undefined') {
+			const brewCollection = this.props.brewCollection.map((brewGroup)=>{
+				const localVisibility = (localStorage.getItem(`${USERPAGE_KEY_PREFIX}-${brewGroup.class}`) ?? 'true')=='true';
+				brewGroup.visible = (brewGroup.visible != localVisibility ? localVisibility : brewGroup.visible);
```
[the remark this answers is not in the document]
Comment (G-Ambatte) [reply client/homebrew/pages/basePages/listPage/listPage.jsx:50]: Looking at this now, it's "if brewGroup.visible doesn't equal localVisibility, set it to localVisibility, but if it does then leave it alone" which always resolves to "set brewGroup.visible to localVisibility". I'll clean it up shortly.

**Comment by G-Ambatte** [comment]
I thought I was committing only one set of changes, however, it was actually both. The latest commit resolves both the unnecessary `visible` prop and the unnecessary logic around setting `brewGroup.visible` in componentDidMount.

File: client/homebrew/pages/basePages/listPage/listPage.less [hunk 3]
```diff
@@ -74,4 +74,27 @@
...
+		&.inactive {
+			color: #707070;
+		}
+		&.inactive::before {
+			content: '\f105';
+			font-family: 'Font Awesome 5 Free';
+			font-weight: 900;
+			font-size: 0.6cm;
+    		padding-right: 0.5em;
+		}
```
Comment (calculuschild) [inline client/homebrew/pages/basePages/listPage/listPage.less:98]: Possible opportunity to reduce duplicated CSS?

```suggestion
		&.active {
			color: #58180D;
		}
		&.inactive {
			color: #707070;
			
		}
		&.active::before, &.inactive::before {
			font-family: 'Font Awesome 5 Free';
			font-weight: 900;
			font-size: 0.6cm;
    			padding-right: 0.5em;
		}
		&.active::before {
			content: '\f107';
		}
		&.inactive::before {
			content: '\f105';
		}
```

File: client/homebrew/pages/userPage/userPage.jsx [hunk 4]
```diff
@@ -31,9 +31,10 @@ const UserPage = createClass({
 
 		const brewCollection = [
 			{
-				title : `${usernameWithS} published brews`,
-				class : 'published',
-				brews : brews.published
+				title   : `${usernameWithS} published brews`,
+				class   : 'published',
+				brews   : brews.published,
+				visible : true
```
Comment (calculuschild) [inline client/homebrew/pages/userPage/userPage.jsx:37]: Can we remove this `visible : true` as well?

**Comment by jeddai** [comment]
Ditto on the other comments, otherwise I tested this and it works great. Do we have any concerns about someone logging out and back in as a different user but seeing the user page collapse status of the previous user?

**Comment by G-Ambatte** [comment]
> Ditto on the other comments, otherwise I tested this and it works great. Do we have any concerns about someone logging out and back in as a different user but seeing the user page collapse status of the previous user?

My first iteration appended the current username to the local storage configuration item, so you only got your own setting. However, I realized that leaking someone's username was far worse than leaking their brew visibility setting, so I removed that concatenation.

Could probably use a hashed version of the username instead, to achieve the same effect.

**Comment by calculuschild** [comment]
Let's keep it as-is for now. We can consider this later on as we get more into personal user account settings.

File: client/homebrew/pages/userPage/userPage.jsx (hunk 4 above)
Comment (G-Ambatte) [reply client/homebrew/pages/userPage/userPage.jsx]: Should be removed in the latest commit.

**Comment by calculuschild** [comment]
Looks good! 💯 🎉 👉😎👉

**Pull request merged**